src/rosenfeld_fmt.cc

```diff
@@ -434,35 +431,69 @@ void RosenfeldFMT::computeSphericalDerivative(std::shared_ptr<State> state)
                 dF_dn3(idx) = 0.;
                 dF_dnv1(idx) = 0.;
                 dF_dnv2(idx) = 0.;
-
+                
+                double dF_dn3_1 = 0;
+                double dF_dn3_2 = 0;
                 // take integrals using quadrature
                 const auto r = mesh->center(idx);
-                const auto lower_bin = mesh->bin(0.);
-                const auto split_bin = mesh->bin(R-r);
-                const auto upper_bin = mesh->bin(r+R);
+                const auto lower = 0.;
+                const auto split = R-r;
+                const auto upper = r+R;
                 const double sq_R = R*R;
-                for (int ig_idx=lower_bin; ig_idx < split_bin; ++ig_idx)
+                
+                double n = 100;
+                double x;
+                double dr;
+                
+                //Trapezoidal integral 1
+                dr = split/n;
+                x = lower;
+                
```

Comment:
Even though `lower = 0`, you should still include it here in case you ever needed to change `lower` for some reason. This also makes the variable declaration more compact:
```suggestion
                // integral 1
                // we have found 100 points works well
                const int n = 100;
                double dr = (split-lower)/n;
                double x = lower;
```